### ISSUE TITLE
Max step size option

### ISSUE THREAD

**Issue by rtimms** [pybamm-team/PyBaMM#2253]
Add a `max_step` parameter to `BaseSolver` and pass it to the solvers in the appropriate way. Then for drive cycle simulations we should set `max_step` to the step size in the data.

**Comment by rtimms** [comment]
Or at least do that in our drive cycle examples

**Comment by jaskiratsingh2000** [comment]
Hey @rtimms What are you excatly expecting out of this? Where can I find this? Thanks!

**Comment by rtimms** [comment]
I wanted to add a `max_step` parameter to `BaseSolver` which controls the maximum step size the solver can take. You'll need to look at the docs for the each of the solvers we use as they may all have different interfaces. E.g. for [scipy integrate](https://docs.scipy.org/doc/scipy/reference/generated/scipy.integrate.solve_ivp.html) you can directly pass `max_step` as an extra option. We should probably follow their syntax so that the default is `inf`, i.e., the step size is not bounded and determined solely by the solver.

Note this is different from `dt_max` used in the `CasadiSolver` to set the step size for event checking. I propose we rename `dt_max` to `dt_event`.

This might be more like medium rather than easy.

**Comment by jaskiratsingh2000** [comment]
I see. Can you provide a basic example for the same because I'm not getting where other parameters are gettiing called in https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/models/submodels/base_submodel.py

**Comment by Saransh-cpp** [comment]
I think you are looking at the wrong class (based on the link you attached). `BaseSolver` -

https://github.com/pybamm-team/PyBaMM/blob/6ab6c15979ad1b11fcebea37e43723179d201ea6/pybamm/solvers/base_solver.py#L19

**Comment by jaskiratsingh2000** [comment]
@rtimms Any update on this?

**Comment by rtimms** [comment]
What do you need an update on? I think you were looking at the wrong class, as suggested by @Saransh-cpp .

**Comment by jaskiratsingh2000** [comment]
@rtimms I was expecting the example on above. That would be helpful if you can provide it.

**Comment by arjxn-py** [comment]
Hi there @rtimms I had gone through the [scipy integrate](https://docs.scipy.org/doc/scipy/reference/generated/scipy.integrate.solve_ivp.html) to try to understand what exacty could be done, but I don't find any parameter like max_step there. Would you like to suggest me anything in regards?

**Comment by medha-14** [comment]
Could I work on this issue if no one else is currently assigned to it?

**Comment by agriyakhetarpal** [comment]
> Could I work on this issue if no one else is currently assigned to it?

Yes – @arjxn-py, if you're not working on #3106 anymore, could @medha-14 try to take over? @medha-14, you can pull that branch from Arjun's fork so that authorship of the previous work is preserved.

**Comment by arjxn-py** [comment]
Hi @medha-14, thank you for extending your interest. Please feel free to take over the PR.
Sorry this one fell through the cracks.
